keras_cv/layers/preprocessing/random_translation_test.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import tensorflow as tf
+from absl.testing import parameterized
+
+from keras_cv.layers.preprocessing.random_translation import RandomTranslation
+
+
+class RandomTranslationTest(tf.test.TestCase, parameterized.TestCase):
```

Comment:
It looks like these tests currently only cover `fill_mode="constant"`.
Let's add some cases for reflect, nearest, and wrap.

Here's some example cases I wrote up (which you'll need to adapt a bit because I was using a different naming convention for the factor params):

```
    def test_translation_with_reflect(self):
        input_image = np.reshape(np.arange(0, 25), (1, 5, 5, 1))
        # Shift by 2 pixels
        layer = preprocessing.RandomTranslation(
            height_factor=(-0.4, -0.4), width_factor=(0, 0), fill_mode="reflect"
        )
        output_image = layer(input_image)
        expected_output = np.reshape(
            np.asarray(
                [
                    [10, 11, 12, 13, 14],
                    [15, 16, 17, 18, 19],
                    [20, 21, 22, 23, 24],
                    [20, 21, 22, 23, 24],
                    [15, 16, 17, 18, 19],
                ]
            ),
            (1, 5, 5, 1),
        )
        self.assertAllEqual(expected_output, output_image)

    def test_translation_with_constant(self):
        input_image = np.reshape(np.arange(0, 25), (1, 5, 5, 1))
        # Shift by 2 pixel
        layer = preprocessing.RandomTranslation(
            height_factor=(-0.4, -0.4),
            width_factor=(0, 0),
            fill_mode="constant",
            fill_value=0,
        )
        output_image = layer(input_image)
        expected_output = np.reshape(
            np.asarray(
                [
                    [10, 11, 12, 13, 14],
                    [15, 16, 17, 18, 19],
                    [20, 21, 22, 23, 24],
                    [0, 0, 0, 0, 0],
                    [0, 0, 0, 0, 0],
                ]
            ),
            (1, 5, 5, 1),
        )
        self.assertAllEqual(expected_output, output_image)

    def test_translation_with_wrap(self):
        input_image = np.reshape(np.arange(0, 25), (1, 5, 5, 1))
        # Shift by 1 pixel
        layer = preprocessing.RandomTranslation(
            height_factor=(-0.4, -0.4), width_factor=(0, 0), fill_mode="wrap"
        )
        output_image = layer(input_image)
        expected_output = np.reshape(
            np.asarray(
                [
                    [10, 11, 12, 13, 14],
                    [15, 16, 17, 18, 19],
                    [20, 21, 22, 23, 24],
                    [0, 1, 2, 3, 4],
                    [5, 6, 7, 8, 9],
                ]
            ),
            (1, 5, 5, 1),
        )
        self.assertAllEqual(expected_output, output_image)

    def test_translation_with_nearest(self):
        input_image = np.reshape(np.arange(0, 25), (1, 5, 5, 1))
        # Shift by 2 pixels
        layer = preprocessing.RandomTranslation(
            height_factor=(-0.4, -0.4), width_factor=(0, 0), fill_mode="nearest"
        )
        output_image = layer(input_image)
        expected_output = np.reshape(
            np.asarray(
                [
                    [10, 11, 12, 13, 14],
                    [15, 16, 17, 18, 19],
                    [20, 21, 22, 23, 24],
                    [20, 21, 22, 23, 24],
                    [20, 21, 22, 23, 24],
                ]
            ),
            (1, 5, 5, 1),
        )
        self.assertAllEqual(expected_output, output_image)
```